### ISSUE TITLE
Different behavior of AudioEngine on different platfroms

### ISSUE THREAD

**Issue by dumganhar** [cocos2d/cocos2d-x#16215]
Currently, audio pause/resume logic on iOS will be controlled automatically.
Both `SimpleAudioEngine` & `New AudioEngine` on iOS achieve in that way, but other platforms are not implemented. So developers have to invoke `AudioEngine::pauseAll/resumeAll` in the Applegate.cpp, some code like:

``` c++
// This function will be called when the app is inactive. Note, when receiving a phone call it is invoked.
void AppDelegate::applicationDidEnterBackground() {
    Director::getInstance()->stopAnimation();

    // if you use SimpleAudioEngine, it must be paused
    SimpleAudioEngine::getInstance()->pauseBackgroundMusic();
    SimpleAudioEngine::getInstance()->pauseAllEffects();
    // if using new audio engine
    AudioEngine::pauseAll();
}

// this function will be called when the app is active again
void AppDelegate::applicationWillEnterForeground() {
    Director::getInstance()->startAnimation();

    // if you use SimpleAudioEngine, it must resume here
    SimpleAudioEngine::getInstance()->resumeBackgroundMusic();
    SimpleAudioEngine::getInstance()->resumeAllEffects();
   // if using new audio engine
   AudioEngine::resumeAll();
}
```

Developers are confused, they complain that why should we invoke these method for Android while their games run perfectly on iOS.

Any thoughts about this? Should we implement the same automatic `pause/resume` strategy for all platforms?

**Comment by minggo** [comment]
@dumganhar does invoke these functions on iOS have any side-effect? If not, i think there is not problem.

**Comment by dumganhar** [comment]
No side effects. But our template comments the code of pause & resume music.
On iOS, developers could leave the template code as it is there, everything will work correctly on iOS/Mac, but it's broken on other platforms.

**Comment by minggo** [comment]
So we should uncomment these codes in template.

**Comment by dumganhar** [comment]
But there is still a problem. Currently, we have two audio engines, One is `SimpleAudioEngine`, other is `cocos2d::experimental::AudioEngine`. 
Would you add pauseAll & resumeAll for both of them?

**Comment by minggo** [comment]
But if we change the behavior on Android, it may break compatibility too. So i think we should just modify the template, add more comments. And i think we should use AudioEngine internally in SimpleAudioEngine, then deprecated SimpleAudioEngine.

**Comment by dumganhar** [comment]
Yep, should this change be in 3.13? If so, please set the Milestone, thanks.

**Comment by minggo** [comment]
done
